x/harvest/keeper/liquidation.go

```diff
@@ -0,0 +1,307 @@
+	}
+
+	return nil
+}
+
+// SeizeDeposits seizes a list of deposits and sends them to auction
+func (k Keeper) SeizeDeposits(ctx sdk.Context, keeper sdk.AccAddress, liqMap map[string]LiqData,
+	deposits []types.Deposit, borrowBalances sdk.Coins, dDenoms, bDenoms []string) error {
+
+	// Seize % of every deposit and send to the keeper
+	aucDeposits := sdk.Coins{}
+	for _, deposit := range deposits {
+		mm, _ := k.GetMoneyMarket(ctx, deposit.Amount.Denom)
+		keeperReward := mm.KeeperRewardPercentage.MulInt(deposit.Amount.Amount).TruncateInt()
+		keeperCoin := sdk.NewCoin(deposit.Amount.Denom, keeperReward)
+
+		// Send keeper their reward
+		err := k.supplyKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, keeper, sdk.NewCoins(keeperCoin))
+		if err != nil {
+			return err
+		}
+
+		// Add remaining deposit coin to aucDeposits
+		aucDeposits = aucDeposits.Add(sdk.NewCoin(deposit.Amount.Denom, deposit.Amount.Amount.Sub(keeperReward)))
+	}
+
+	// Build map to hold deposit coin USD valuations
+	totalRemainingDepositedUSDAmount := sdk.ZeroDec()
+	dUsdMap := make(map[string]sdk.Dec)
+	for _, deposit := range aucDeposits {
+		dData := liqMap[deposit.Denom]
+		dCoinUsdValue := sdk.NewDecFromInt(deposit.Amount).Quo(sdk.NewDecFromInt(dData.conversionFactor)).Mul(dData.price)
+		totalRemainingDepositedUSDAmount = totalRemainingDepositedUSDAmount.Add(dCoinUsdValue)
+		dUsdMap[deposit.Denom] = dCoinUsdValue
+	}
+
+	// Build map to hold borrow coin USD valuations
+	totalBorrowedUSDAmount := sdk.ZeroDec()
+	bUsdMap := make(map[string]sdk.Dec)
+	for _, bCoin := range borrowBalances {
+		bData := liqMap[bCoin.Denom]
+		bCoinUsdValue := sdk.NewDecFromInt(bCoin.Amount).Quo(sdk.NewDecFromInt(bData.conversionFactor)).Mul(bData.price)
+		totalBorrowedUSDAmount = totalBorrowedUSDAmount.Add(bCoinUsdValue)
+		bUsdMap[bCoin.Denom] = bCoinUsdValue
+	}
+
+	// The % by which the lot must be larger than the borrow
+	ltv := totalBorrowedUSDAmount.Quo(totalRemainingDepositedUSDAmount)
+
+	err := k.StartAuctions(ctx, deposits[0].Depositor, bDenoms, dDenoms, borrowBalances, aucDeposits, ltv, liqMap, bUsdMap, dUsdMap)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// StartAuctions attempts to start auctions for seized assets
+func (k Keeper) StartAuctions(ctx sdk.Context, borrower sdk.AccAddress, borrowDenoms, depositDenoms []string,
+	borrows, deposits sdk.Coins, ltv sdk.Dec, liqMap map[string]LiqData, bUsdMap, dUsdMap map[string]sdk.Dec) error {
+	// Set up auction constants
+	returnAddrs := []sdk.AccAddress{borrower}
+	weights := []sdk.Int{sdk.NewInt(100)}
+	debt := sdk.NewCoin("debt", sdk.ZeroInt())
+
+	// 1. Attempt auctions where we can sell all of a borrowed asset type at once
+	for _, bDenom := range borrowDenoms {
+		bCoin := sdk.NewCoin(bDenom, borrows.AmountOf(bDenom))
+
+		for _, dDenom := range depositDenoms {
+			// Search for a deposit coin amount with USD valuation >= desired lot size USD valuation
+			lotSizeUSD := bUsdMap[bDenom].Mul(ltv)
```

Comment:
Fiarly sure this should be `Quo` and not `Mul` - ie if the borrow is worth $80, we want the lot to be 80/0.8 = 100, not 80*0.8 = 64